src/custom-elements.js

```diff
@@ -451,6 +459,9 @@ let Deferred;
         const walker = createTreeWalker(root);
         do {
           const node = /** @type {!HTMLElement} */ (walker.currentNode);
+          if (node.getAttribute('is')) {
```

Comment:
I'm not super familiar with this polyfill, but if it hooks element creation it's more accurate to cache 'is' at creation time. The HTML spec calls this the element's "is value" IIRC.

I guess you probably need to look at attributes for parsed elements.

You may want to keep those values in a side WeakMap of Element -> string. That will avoid the closure type problem you mention below.